completion.go

```diff
@@ -241,6 +256,37 @@ func validateRequestForO1Models(request ChatCompletionRequest) error {
 	return nil
 }
 
+// validateRequestForO3Models checks for deprecated fields of OpenAI models.
+func validateRequestForO3Models(request ChatCompletionRequest) error {
+	if _, found := O3SeriesModels[request.Model]; !found {
+		return nil
+	}
+
+	// Logprobs: not supported.
+	if request.LogProbs {
+		return ErrO3BetaLimitationsLogprobs
+	}
+
+	// Other: temperature, top_p and n are fixed at 1, while presence_penalty and frequency_penalty are fixed at 0.
+	if request.Temperature > 0 && request.Temperature != 1 {
+		return ErrO3BetaLimitationsOther
+	}
+	if request.TopP > 0 && request.TopP != 1 {
+		return ErrO3BetaLimitationsOther
+	}
+	if request.N > 0 && request.N != 1 {
+		return ErrO3BetaLimitationsOther
+	}
+	if request.PresencePenalty > 0 {
+		return ErrO3BetaLimitationsOther
+	}
+	if request.FrequencyPenalty > 0 {
+		return ErrO3BetaLimitationsOther
+	}
```

Comment:
I suggest we mark `ErrO1BetaLimitations*` as deprecated and introduce a single set of errors and a single validation function. Currently, it feels like a pure duplication

Reply:
Thanks, will submit this patch later today (as in, in next few hours)

Reply:
The only issue with this is some components (eg. Function calling) are supported in O3, but not O1

Going to keep those earlier errors (and tests) around - but will remove the rest in favour of O3

Reply:
Done 👍